telemetry/wrapper.go

```diff
@@ -68,6 +72,17 @@ func SetGaugeWithLabels(keys []string, val float32, labels []metrics.Label) {
 
 // MeasureSince provides a wrapper functionality for emitting a a time measure
 // metric with global labels (if any).
-func MeasureSince(start time.Time, keys ...string) {
+func MeasureSince(ctx sdk.Context, start time.Time, keys ...string) {
+	if !ctx.TelemetryEnabled() {
+		return
+	}
 	metrics.MeasureSinceWithLabels(keys, start.UTC(), globalLabels)
 }
+
+// Now return the current time if telemetry is enabled or a zero time if it's not
+func Now(ctx sdk.Context) time.Time {
```

Comment:
as a heads up we want to avoid using sdk.Context anywhere

Reply:
Perfect! This PR is part of [this comment](https://github.com/cosmos/cosmos-sdk/issues/10245#issuecomment-2020642779) I made on this issue: https://github.com/cosmos/cosmos-sdk/issues/10245